src/main/java/com/github/Hyun_jun_Lee0811/dictionary/service/WordService.java

```diff
@@ -0,0 +1,62 @@
+package com.github.Hyun_jun_Lee0811.dictionary.service;
+
+import com.github.Hyun_jun_Lee0811.dictionary.client.WordnikClient;
+import com.github.Hyun_jun_Lee0811.dictionary.model.dto.*;
+import lombok.RequiredArgsConstructor;
+import org.apache.commons.collections4.Trie;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
+public class WordService {
+
+  private final WordnikClient wordnikClient;
+  private final Trie<String, Boolean> trie;
+
+  public List<WordDefinitionDto> getDefinitions(String word) {
+    List<WordDefinitionDto> definitions = wordnikClient.getDefinitions(normalizeWord(word));
+    updateTrie(normalizeWord(word));
+    return definitions;
+  }
+
+  public WordExampleDto getExamples(String word) {
+    WordExampleDto examples = wordnikClient.getExamples(normalizeWord(word));
+    updateTrie(normalizeWord(word));
+    return examples;
+  }
+
+  public List<WordRelatedWordDto> getRelatedWords(String word) {
+    List<WordRelatedWordDto> relatedWords = wordnikClient.getRelatedWords(normalizeWord(word));
+    updateTrie(normalizeWord(word));
```

Comment:
normalizeWord(word)가 계속 반복적으로 들어가 있네요. 다음과 같이 선언하여 사용하시면 더 좋을것 같습니다
```suggestion
    String normalizedWord = normalizeWord(word)
    List<WordRelatedWordDto> relatedWords = wordnikClient.getRelatedWords(normalizedWord);
    updateTrie(normalizedWord);
```

Reply:
@mingming-mentor
Elasticsearch와 연동하였습니다.
ElasticSearchConfigTest를 작성하여 Elasticsearch 클라이언트 구성 검증과 Elasticsearch 클러스터와의 연결에 관한 테스트도 작성하였습니다!